crates/web5/src/apid/dids/methods/did_dht/mod.rs

```diff
@@ -44,14 +115,117 @@ impl DidDht {
     }
 
     pub fn resolve(uri: &str) -> Result<ResolutionResult> {
```

Comment:
I appreciate the staying-true to Rust idioms with the return value of `Result<ResolutionResult>` here, but to be conformant to the spec, we should return `ResolutionResult` and use the embedded `ResolutionMetadataError` for error cases (those which are specific to rust runtime errors such as json serialization, etc. would be `ResolutionMetadataError::InternalError`. It's a bit awkward, because we follow the monad pattern with `Result<T>` everywhere, but yeah we should be conformant to the spec and utilize the embedded `error` data member. 

You can see how I did this over in `did:jwk` using a closure (admittedly it may not be exactly the same here because we still want to capture the `ResolutionMetadataError` variants in the places you're early-returning via the `?` -- as opposed to in `did:jwk` it all maps to the `InternalError` variant.

I can take a stab at this @diehuxx and you can review and lmk if you think I did things incorrectly.

Reply:
Okay done here, lmk if you take issue @diehuxx https://github.com/TBD54566975/web5-rs/pull/250/commits/dff3fa72927ecc44868a49d89dd5ded88b2c27ec

Also it made me realize I needed to improve the error cases over in the did:jwk resolve()

Reply:
Also realizing I need to fix did:web accordingly too, but will do that in a different PR